src/core/config/upstream.rs

```diff
@@ -28,7 +28,13 @@ impl Default for Batch {
         }
     }
 }
-
+#[derive(Serialize, Deserialize, PartialEq, Eq, Clone, Debug, schemars::JsonSchema, MergeRight)]
+#[serde(rename_all = "camelCase", default)]
+#[derive(Default)]
+pub struct HttpCache {
+    pub enable: bool,
+    pub size: u64,
```

Comment:
now that we are taking `size` we do not need `enable`.  If user  specifies size, it's obvious to turn httpCache on.

and secondly, you should skip serialization if default

Reply:
@ssddOnTop _**enable**_ was suggested by @tusharmath  if user wants _**size**_ to represent percentage rather than number of entries example below.

```graphql
schema
  @server(port: 4000)
  @upstream(
    baseURL: "https://api.example.com"
    # highlight-start
    httpCache: {size: 42} # Fixed number of entries
    # OR
    httpCache: {enable: true, size: 2} # 2% of available memory (i.e. RAM)
    # highlight-end
  ) {
  query: Query
}
```